akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -239,6 +239,63 @@ private[akka] final case class Fold[In, Out](zero: Out, f: (Out, In) ⇒ Out, de
 /**
  * INTERNAL API
  */
+private[akka] final case class Intersperse[T](start: Option[T], inject: T, end: Option[T]) extends StatefulStage[T, T] {
```

Comment:
No emit, but I found it a bit overkill to have an Iterator/Iterable for maximum two elem output sequences.

You can have a two-elem array, a `read` index and an `end` index. Then you only need the code to emit at most two length sequences:

[start], elem1
sep, elem2
sep, elem3
sep, elem4
[end]

maybe overkill but I think it is simple enough.